consumer/consumer.go

```diff
@@ -3,177 +3,193 @@ package consumer
 import (
 	"context"
 	"fmt"
-	"io/ioutil"
-	"log"
+	"sync"
 	"time"
 
 	"github.com/Shopify/sarama"
 	"github.com/heetch/felice/codec"
 	"github.com/pkg/errors"
-	"gopkg.in/retry.v1"
 )
 
-// Consumer is the structure used to consume messages from Kafka.
-// Having constructed a Consumer with New, you should use its Handle method to
-// register per-topic handlers and, finally, call it's Serve method to
-// begin consuming messages.
+// Consumer represents a Kafka message consumer.
+// See the package documentation and New for
+// information on how to use it.
 type Consumer struct {
-	// Metrics stores a type that implements the MetricsReporter interface.
-	// If you provide an implementation, then its Report function will be called every time
-	// a message is successfully handled.  The Report function will
-	// receive a copy of the message that was handled, along with
-	// a map[string]string containing metrics about the handling of the
-	// message.  Currently we pass the following metrics: "attempts" (the
-	// number of attempts it took before the message was handled);
-	// "msgOffset" (the marked offset for the message on the topic); and
-	// "remainingOffset" (the difference between the high water mark and
-	// the current offset).
-	Metrics MetricsReporter
+	config Config
+
+	// mu guards the fields below.
+	mu sync.Mutex
 
-	// Logger output defaults to ioutil.Discard.
-	// If you wish to provide a different value for the Logger, you must do this prior to calling Consumer.Serve.
-	Logger *log.Logger
+	// quit is closed when the consumer is closed.
+	quit chan struct{}
 
-	consumer sarama.ConsumerGroup
-	config   *Config
-	handlers *collection
-	quit     chan struct{}
+	// consumerGroup holds the consumerGroup started by Serve.
+	consumerGroup sarama.ConsumerGroup
+	// handlers holds the set of handlers.
```

Comment:
This comment isn't really adding much, IMHO.

Reply:
removed